search/vertex-search-options/vertex_search_options.ipynb

```diff
@@ -0,0 +1,649 @@
+        "</CONTEXT>\n",
+        "\n",
+        "<CONSTRAINTS>\n",
+        "- ONLY use information available from the Grounding tool\n",
+        "</CONSTRAINTS>\n",
+        "\n",
+        "<OUTPUT_FORMAT>\n",
+        "- Response should ALWAYS be in following JSON Output with answer and reference_snippet as keys, e.g. {\"answer\": , \"reference_snippet\": }\n",
+        "</OUTPUT_FORMAT>\n",
+        "\"\"\""
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "wSo25gr9MBSd"
+      },
+      "outputs": [],
+      "source": [
+        "tools = [\n",
+        "    Tool.from_retrieval(\n",
+        "        retrieval=generative_models.grounding.Retrieval(\n",
+        "            source=generative_models.grounding.VertexAISearch(\n",
+        "                datastore=\"projects/801452371447/locations/global/collections/default_collection/dataStores/alphabet-contracts1-id\"\n",
+        "            ),\n",
+        "            disable_attribution=False,\n",
+        "        )\n",
+        "    ),\n",
+        "]\n",
+        "\n",
+        "generation_config = {\n",
+        "    \"max_output_tokens\": 8192,\n",
+        "    \"temperature\": 1,\n",
+        "    \"top_p\": 0.95,\n",
+        "}\n",
+        "\n",
+        "safety_settings = [\n",
+        "    SafetySetting(\n",
+        "        category=SafetySetting.HarmCategory.HARM_CATEGORY_HATE_SPEECH,\n",
+        "        threshold=SafetySetting.HarmBlockThreshold.BLOCK_ONLY_HIGH,\n",
+        "    ),\n",
+        "    SafetySetting(\n",
+        "        category=SafetySetting.HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT,\n",
+        "        threshold=SafetySetting.HarmBlockThreshold.BLOCK_ONLY_HIGH,\n",
+        "    ),\n",
+        "    SafetySetting(\n",
+        "        category=SafetySetting.HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT,\n",
+        "        threshold=SafetySetting.HarmBlockThreshold.BLOCK_ONLY_HIGH,\n",
+        "    ),\n",
+        "    SafetySetting(\n",
+        "        category=SafetySetting.HarmCategory.HARM_CATEGORY_HARASSMENT,\n",
+        "        threshold=SafetySetting.HarmBlockThreshold.BLOCK_ONLY_HIGH,\n",
+        "    ),\n",
+        "]"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "jCJAOvHdLrGI"
+      },
+      "outputs": [],
+      "source": [
+        "def generate():\n",
+        "    model = GenerativeModel(\n",
+        "        \"gemini-1.5-pro-001\", tools=tools, system_instruction=[system_instructions]\n",
+        "    )\n",
+        "\n",
+        "    response = model.generate_content(\n",
+        "        [SEARCH_QUERY],\n",
+        "        generation_config=generation_config,\n",
+        "        safety_settings=safety_settings,\n",
+        "        stream=True,\n",
+        "    )\n",
+        "\n",
+        "    return response"
```

Comment:
Is there a need to wrap this in a function?

Reply:
thought it was neater, but you are right since I dont reuse it in this notebook, have updated the code